@here/harp-geoutils/lib/projection/Projection.ts

```diff
@@ -123,43 +123,93 @@ export interface Projection {
      * meters.
      *
      * @param worldPoint The position in world coordinates.
      */
-    getScaleFactor(worldPoint: Vector3Like): number;
+    abstract getScaleFactor(worldPoint: Vector3Like): number;
 
     /**
      * Returns the surface normal at the given world position.
      *
      * @param worldPoint The position in world coordinates.
      */
-    surfaceNormal(worldPoint: Vector3Like): Vector3Like;
+    abstract surfaceNormal(worldPoint: Vector3Like): Vector3Like;
 
     /**
      * Returns the surface normal at the given world position.
      *
      * @param worldPoint The position in world coordinates.
      * @returns The resulting normal vector.
      */
-    surfaceNormal<Normal extends Vector3Like>(worldPoint: Vector3Like, result: Normal): Normal;
+    abstract surfaceNormal<Normal extends Vector3Like>(
+        worldPoint: Vector3Like,
+        result: Normal
+    ): Normal;
 
     /**
      * Returns the signed distance between the given coordinates and
      * the closest point on the surface.
      *
      * @param worldPoint The position in world coordinates.
      */
-    groundDistance(worldPoint: Vector3Like): number;
+    abstract groundDistance(worldPoint: Vector3Like): number;
 
     /**
      * Scales the given world coordinates to the surface.
      *
      * @param worldPoint The position in world coordinates.
      */
-    scalePointToSurface(worldPoint: Vector3Like): Vector3Like;
+    abstract scalePointToSurface(worldPoint: Vector3Like): Vector3Like;
+
+    /**
+     * Reproject a world position from the given source [[Projection]].
+     *
+     * @param sourceProjection The source projection.
+     * @param worldPos A valid world position for the given source projection.
+     * @returns The world position reprojected using this [[Projection]].
+     */
+    reprojectPoint(sourceProjection: Projection, worldPos: Vector3Like): Vector3Like;
+
+    /**
+     * Reproject a world position from the given source [[Projection]].
+     *
+     * @param sourceProjection The source projection.
+     * @param worldPos A valid position in the world space defined by the source projection.
+     * @param result The resulting position reprojected using this [[Projection]].
+     */
+    reprojectPoint<WorldCoordinates extends Vector3Like>(
+        sourceProjection: Projection,
+        worldPos: Vector3Like,
+        result: WorldCoordinates
+    ): typeof result;
+
+    /**
+     * Reproject a world position from the given source [[Projection]].
+     *
+     * @param sourceProjection The source projection.
+     * @param worldPos A valid position in the world space defined by the source projection.
+     * @param result The resulting position reprojected using this [[Projection]].
+     * @hidden
+     */
+    reprojectPoint(
+        sourceProjection: Projection,
+        worldPos: Vector3Like,
+        result?: Vector3Like
+    ): Vector3Like {
+        if (sourceProjection === this) {
```

Comment:
super edge case: should you check instanceof as well?

Reply:
there shouldn't be any need for checking with `instanceof`. Our projections are implemented as singleton (and the classes are not exported, well MercatorProjection is exported but that is only because WebMercator subclasses it). By the way the only exported symbols are `mercatorProjection, webMercatorProjection, identityProjection, sphereProjection` and so on..). So there should be one and only one instance of each projection so the strict equality check should be enough.